sites/main-site/src/content/blog/2024/seqera-containers-part-2.mdx

```diff
@@ -0,0 +1,646 @@
+```
+
+:::
+
+## Pipelines
+
+Container information at shared module-level is great, but it's not enough.
+Nextflow doesn't know about module `meta.yml` files (they're an nf-core invention),
+so we need to tie these into the pipeline code where they will run.
+
+The heart of the solution is to auto-generate a config file for each software packaging type (Docker, Singularity, Conda)
+and platform (`linux/arch64` and `linux/arm64`).
+These will be created by the nf-core/tools CLI and never be edited by hand, so no manual merging will be required.
+They'll simply be regenerated and overwritten every time a version of a module is updated.
+
+Each config file will specify the `container` or `conda` directive for every process in the pipeline:
+
+```groovy title="config/containers_docker_amd64.config"
+// AUTOGENERATED CONFIG BELOW THIS POINT - DO NOT EDIT
+process { withName: 'NF_PIPELINE:FASTQC'         { container = 'fastqc:0.12.1--5cfd0f3cb6760c42' } }
+process { withName: 'NF_PIPELINE:MULTIQC'        { container = 'multiqc:1.25--9968ff4994a2e2d7' } }
+process { withName: 'NF_PIPELINE:ANALYSIS_PLOTS' { container = 'express_click_pandas_plotly_typing:58d94b8a8e79e144' } }
+//.. and so on, for each process in the pipeline
+```
+
+Likewise, the conda config files will point to the lockfiles for each process:
+
+```groovy title="config/conda_lockfiles_amd64.config"
+// AUTOGENERATED CONFIG BELOW THIS POINT - DO NOT EDIT
+process { withName: 'NF_PIPELINE:FASTQC'         { conda = 'https://wave.seqera.io/v1alpha1/builds/5cfd0f3cb6760c42_1/condalock' } }
+process { withName: 'NF_PIPELINE:MULTIQC'        { conda = 'https://wave.seqera.io/v1alpha1/builds/9968ff4994a2e2d7_1/condalock' } }
+process { withName: 'NF_PIPELINE:ANALYSIS_PLOTS' { conda = 'https://wave.seqera.io/v1alpha1/builds/58d94b8a8e79e144_1/condalock' } }
+//.. and so on, for each process in the pipeline
+```
+
+The main `nextflow.config` file will import these config files, depending on the [profile selected](#usage-summary)
+by the person running the pipeline.
+
+Singularity will have separate config files and associated `-profile`s for both `oras` and `https` containers,
+so that users can choose which to use.
+
+Local modules and any edge-case shared modules that cannot use the Seqera Containers automation
+will need the pipeline developer to hardcode container names and conda lockfiles manually.
+These can be added to the above config files as long as they remain above the comment line:
+
+```groovy
+// AUTOGENERATED CONFIG BELOW THIS POINT - DO NOT EDIT
+```
+
+We're also taking this opportunity to update the `apptainer` and `mamba` profiles too,
+they will import the exact same config files as the `singularity` and `conda` profiles.
+
+Here's roughly how the `nextflow.config` file with the `-profile` config includes will look:
+
+::::info{.fa-code title="nextflow.config" collapse}
+
+:::note
+Boilerplate code (eg. disabling other container engines) has been removed from this
```

Comment:
Hmmm should definitely have the disabling of the others, we implemented because @maxibor found a really awkward behaviour when we didn't do this a while ago (don't have time to find that thread atm though)

Reply:
Yup, I only mean for the sake of the blog post. I'll make this more clear.

Reply:
Done in 8c2cab641